### DISCUSSION TITLE
主题未来的计划和一些建议

### DISCUSSION THREAD

**Discussion by liuyib** [liuyib/hexo-theme-stun#10] [Show and tell]
目前主题的功能已经满足了我的需求，因此基本不会再进行新功能的开发，但会一直维护，修复 Bug。如果需要添加新功能，欢迎通过 Pull Requests 贡献代码。🤗

---

由于 Github 推出了 [讨论功能](https://github.com/liuyib/hexo-theme-stun/discussions)，类似于聊天室，因此本项目以后的交流解答都会在其中进行。原来的 [Gitter](https://gitter.im/liuyib/hexo-theme-stun) 聊天室作废，不再使用。QQ 群也不再建议加入了（有可能会解散）。

如果你在犹豫，是提出 Issues 合适，还是新建 Discussions 更合适，那么我的建议是：**如果是使用上的问题（无法启动、报错、等等）请在 Discussions 中新建一个讨论。如果你感觉是主题 Bug 或有功能性的建议，请务必按照模板提出 [Issues](https://github.com/liuyib/hexo-theme-stun/issues)。如果你还是不确定哪个更合适，那就随意挑选一个，抛硬币也行 `:)`**。

---

一些建议：

- 使用 Gitalk 评论时，如果文章路径存在中文，可能出现评论丢失的问题。因此建议将文章的 Markdown 文件用英文命名，然后将 Front-Matter 中的 `title` 设为文章的中文名字。例如：
  
  ![TIM图片20190923163113](https://user-images.githubusercontent.com/38221479/65411602-d2976380-de1f-11e9-802e-51c80707f35e.png)

  这样做额外的好处是，即使你修改了文章标题，文章的链接也不会 404 👏。

- `...`

---

下面的评论是之前 Issues 中旧的评论，如果有问题，不要再在下面评论了，请直接新建 Discussion。

**Comment by yixvan6** [comment]
你好，可以添加背景音乐吗

**Comment by liuyib** [comment]
@yixvan6 你好。我的想法是博客应该专注于内容，因此辅助类的功能不会支持太多。

比如你说的背景音乐，讲道理这个功能不属于主题的范畴，所以最好是通过插件来提供这个功能。

既然提到了，后面会考虑通过插件来进行支持该功能。

**Comment by ghost** [comment]
可以添加发布视频的功能吗

**Comment by liuyib** [comment]
@1565031532 你的意思是在文章中嵌入本地视频或者在线的视频链接吗？如果是的话，暂不支持嵌入本地视频。不过 Hexo 支持嵌入在线视频。比如，你可以使用 iframe 嵌入：

```js
{% iframe //player.bilibili.com/player.html?aid=370010949&bvid=BV1NZ4y1j7nw&cid=172423516&page=1 %}
```

或者是，通过 [`{% raw %}{% endraw %}`](https://hexo.io/zh-cn/docs/tag-plugins#Raw) 嵌入：

```js
{% raw %}
<iframe src="//player.bilibili.com/player.html?aid=370010949&bvid=BV1NZ4y1j7nw&cid=172423516&page=1" scrolling="no" border="0" frameborder="no" framespacing="0" allowfullscreen="true"> </iframe>
{% endraw %}
```

**Comment by liuyib** [comment]
@1565031532 Hexo 提供的 `{% raw %}{% endraw %}` 标签，就是用来嵌入源代码，所以你可以利用这个干任何事情（在文章的 Markdown 文件中嵌入任何代码）。

我试了下，利用这个功能，可以插入本地视频。举例如下：

> 利用 HTML5 的 `video ` 标签

```html
{% raw %}
<video controls width="250">
  <!-- 把视频资源放在 Hexo 的某个目录（例如 /assets 目录）下，然后再引入 -->
  <source src="/assets/1.mp4">
  Sorry, your browser doesn't support embedded videos.
</video>
{% endraw %}
```

插入在线视频，就是我上面提到的那样。

**Comment by liuyib** [comment]
@gongdananyou 指定某个页面取消 sidebar 的功能已添加，详见：releases: [Stun v2.1.1](https://github.com/liuyib/hexo-theme-stun/releases/tag/2.1.1)，commit: [cf31687](https://github.com/liuyib/hexo-theme-stun/commit/cf31687e2b12ab37118ebd0d750f07a98fd451d6)

**Comment by Shaun-Shane** [comment]
你好，请问可以添加代码折叠功能吗？

**Comment by liuyib** [comment]
> 你好，请问可以添加代码折叠功能吗？

你好，这个可以支持，未来的版本中支持该功能之后，会在这里通知你 🙌

**Comment by JunKuangKuang** [comment]
请问可以考虑在侧边栏添加天气插件么?
https://cj.weather.com.cn/plugin/standard

**Comment by liuyib** [comment]
@Keen-King  这个暂时不考虑支持，我目前在重构 Stun，等到 3.0 版本的时候，会支持通过配置指定侧边栏模板（一种改源码比较优雅的方式）。目前的话，你可以先修改源码实现。或者去找一下，有没有别人写好的相关插件（配置之后一键使用的那种）。

**Comment by catch6** [comment]
文章页头部的发表于，更新于，阅读量等的文字颜色可以默认为color: var(--color-gray-700);吗，目前看起来颜色太跳不协调，而且转移了读者的注意力，或者设置成可配置的

**Comment by liuyib** [comment]
> 文章页头部的发表于，更新于，阅读量等的文字颜色可以默认为color: var(--color-gray-700);吗，目前看起来颜色太跳不协调，而且转移了读者的注意力，或者设置成可配置的

你说的这个，可以通过覆盖 CSS 样式实现，看下文档[这里](https://theme-stun.github.io/docs/zh-CN/advanced/advanced.html#%E8%87%AA%E5%AE%9A%E4%B9%89%E6%A0%B7%E5%BC%8F)。具体做法就是，你在网页上，查看源码，找到要修改元素的样式，然后按照文档里的方法，在指定的文件中修改就好了。
